packages/nimble-components/src/unit/byte/models/tests/byte-1024-unit-scale-formatter.spec.ts

```diff
@@ -0,0 +1,63 @@
+import { Byte1024UnitScaleFormatter } from '../byte-1024-unit-scale-formatter';
+
+describe('Byte1024UnitScaleFormatter', () => {
```

Comment:
> Could we add cases beyond the range of PiB like you did for volts?

Yes. Added.

> do we have unit-agnostic test coverage that ensures invalid values formatted as units display correctly?

Kind of. With or without units, the number portion gets formatted the same way. We have existing tests (for the DecimalFormatter and DefaultFormatter) that validate the formatting of interesting values, like NaN, -0, Infinity, etc. But none of those values format as an empty string. So if you format NaN without a unit, it's "NaN"; with the bytes unit, it's "NaN bytes". But we don't explicitly test for the latter, since the unit's presence doesn't affect the formatting of the number.

Reply:
How about the cases that result in an empty number like `undefined` and `null`? Will those show " bytes" or empty string? I'd advocate for the latter (and for test coverage that validates it).

Reply:
Yes, that is handled just by deriving from `NumberFormatter`, but I've added testing for it to `unit-scale-formatter.spec.ts` anyway.